### ISSUE TITLE
Add CrudRepository

### ISSUE THREAD

**Issue by andrejromanov** [WinterFramework/winter#140]
Currently Winter doesn't have entities repositories. It's time to add them!
We should add CrudRepository which will implement standard repository methods:
- create(factory)
- save(aggregate_instance)
- delete(aggregate_instance)
- find_by_id(id)

**Comment by berenda-andrey** [comment]
@andrejromanov Seems that winter can only give abstract class and after that you can inherit from it and use. And  abstract class can only restrict you without giving something useful(inheriting is not so good in python). if you think different please let me know.
I think we need think about integreting repositories with main winter feature like so:

```
class UserController:

    @use_repository(UserRepository, identifier='id')
    @route_get('/users/{id}/')
    def get_user(user: User): # getting user instead of id
        return UserDTO(user.id, user.name)
```
(I think it is bad example but for showing idea is ok)
Without integreting repositories you can use other packages without having any restricts in winter

**Comment by mofr** [comment]
@berenda-andrey Hello! Agree with you about abstract classes. SQLAlchemy implementation would be implemented in another library `winter-sqlalchemy` which could be included in this repository. Also, I think we should support user-defined methods in repositories, and I don't have any good ideas in this direction at the moment.


Also, I think that repository linking with an entity class should be done during declaration, for example:
```
from winter.persistense import CRUDRepository

class UserRepository(CRUDRepository[User]):
    pass
    # custom methods here, don't know in what form yet
```

**Comment by mofr** [comment]
The next logical step is to incorporate [persipy](https://github.com/pristupa/persipy/) and [slqar](https://github.com/pristupa/sqlar/) into `winter.persistense` and `winter.persistence-sqlalchemy`.

**Comment by mofr** [comment]
Basic repository functionality is already in master and is planned to be released with Winter 3.0.
The current implementation lacks: custom methods implementation, pagination, domain events.
